internal/wasm/wasm.go

```diff
@@ -42,6 +42,10 @@ type InitMetadata struct {
 	//   - Random number generators are seeded with a deterministic source.
 	StrictSandbox bool `mapstructure:"strictSandbox"`
 
+	// GuestConfig is the guest configuration, assigned during instantiation by handler.GuestConfig.
+	// Currently only used by the HTTP wasm middleware.
+	GuestConfig string `mapstructure:"guestConfig"`
```

Comment:
ok with me, though I think a follow-up we can try to refactor so that base metadata can be parsed without adding uncommon fields.

Reply:
Done, reverted the change to the existing binaries, and also moved GuestConfig to the wasm middleware package.